### DISCUSSION TITLE
Creating binding generators / cleaner wasm compiles.

### DISCUSSION THREAD

**Discussion by crystalthoughts** [grame-cncm/faust#921] [General]
Hi - 
I love writing Faust but I find the wasm tools to generate a lot of unwanted code if just an API to a web audio worker is desired.
 I'd like really just to load the wasm file and have a single getter/setter per parameter. Do you have any advice regarding this or is there a writeup I'm missing? 
At one point I found myself having to cut out a javascript module that is passed as a string for some reason which I'm sure doesn't need to happen that way :)

I would also be interested in creating wasm wrappers for other runtimes like WASIX.

**Comment by sletz** [comment]
Faust and Web tools are under heavy rework, you can have a look of the WIP here: https://www.npmjs.com/package/@grame/faustwasm

**Comment by crystalthoughts** [comment]
Thanks I’ll take a look
